python/ray/air/checkpoint.py

```diff
@@ -454,6 +468,34 @@ def from_directory(cls, path: str) -> "Checkpoint":
 
         return checkpoint
 
+    @classmethod
+    def from_directory_populator(cls, populator: Callable[[str], Any]):
```

Comment:
As a user it's actually not clear that this is a temporary or managed checkpoint?

To @Yard1's point I'm actually wondering if we want this to only exist within the context of a `Session`, e.g. create a `SessionManagedCheckpoint`, which we would allow to handle:
- "temporary" checkpoints that are specifically persisted with `session.report(metrics, checkpoint)`
- "managed" checkpoints that can be stored at a path that is specific to the session.

Reply:
It's not a temporary or managed checkpoint. It's just a checkpoint.

If you deconstruct a dict checkpoint, the memory is freed. If you deconstruct this file checkpoint, the disk is freed. If you send it somewhere else first, it may or may not be persisted (same with dict checkpoints).

We should not think of Checkpoints as being different depending on their internal data representation. When I send a Checkpoint via the object store, it's actually a bytes checkpoint on deserialization. We don't care as long as we can convert it into a processable format with to_dict() or to_directory().

The fact that this checkpoint is actually stored on disk is an implementation detail. It could also live in memory.

Reply:
Gotcha, that logic makes sense. I think I was confused by the naming of the `_managed` attribute... 

On a tangent, it probably would be helpful if we had a diagram/documentation to outline the possible lifecycles of a Checkpoint.

Reply:
Should we rename it to e.g. `_ephemeral`?